### ISSUE TITLE
Beaglebone Black rootfs hangs

### ISSUE THREAD

**Issue by GithubIsAwful** [void-linux/void-packages#2238]
The boot process for the Beaglebone Black build hangs, presumably an issue with the kernel. If there was a way to verbosely print the boot process with U-Boot I would try to solve this further than I have. This is the tail of the output over a serial connection.

=> Setting up hostname to 'void-live'...
=> Loading kernel modules...

=> Loading sysctl(8) settings...

Applying /usr/lib/sysctl.d/void.conf ...
kernel.core_uses_pid = 1
fs.protected_hardlinks = 1
fs.protected_symlinks = 1
Applying /etc/sysctl.conf ...
=> Initialization complete, running stage 2...
runit: leave stage: /etc/runit/1
runit: enter stage: /etc/runit/2
runsvchdir: default: current.
[ 9.328666] udevd[212]: starting version 3.1.5
[ 9.563969] net eth0: initializing cpsw version 1.12 (0)
[ 9.571781] net eth0: phy found : id is : 0x7c0f1
[ 9.576769] libphy: PHY 4a101000.mdio:01 not found
[ 9.581805] net eth0: phy 4a101000.mdio:01 not found on slave 1
[ 9.596308] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
[ 12.650271] libphy: 4a101000.mdio:00 - Link is Up - 100/Full
[ 12.656351] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready

I have tried disabling IPv6 in systemctl.conf (which appears to do nothing, the IPv6 ADDRCONF messages still appear), and have tried disabling udev with `touch etc/runit/runsvdir/current/udevd/down` which also appears to do nothing.

**Comment by daisybytes** [comment]
I recently tried the same and also ran into this issue.

I used void-mklive to create a fresh rootfs but that didn't boot up at all. U-Boot died with a "file not found" error looking for am335x-boneblack.dts (if I remember correctly now). The created image does included this within boot/dtbs (also did some quick research and it looked like thats where it should be placed) but only if I moved the file a level higher than its current position inside dtbs the beagle started booting.

After the network interface comes up it doesnt aquire a lease from DHCP and just gets stuck.
I also tried removing services, but nothing helped :(

**Comment by daisybytes** [comment]
Digged a little more.

Disabling IPV6 makes no difference. I attached a USB Ethernet dongle to see if the issue lies within the internal network adapter but I cannot connect the Beagle that way neither.

By enabling the sulogin service I was able to login to the machine after boot and set the internal ethernet adapter to a static address and fire up ntpd. I can pink the void repository but immediately after running "xbps-update -S" that process gets "Killed". Unfortunately nothing gets logged and the rootfs has no tracing capabilities. Checking available memory shows plenty (about 95% or more) so I'm really puzzled.

I noticed that the rootfs seems to have no uboot.env, at least when booting no environment gets used and the Beagle tells me its falling back to default. Maybe it needs some specific settings here without which the memory gets seriously f*ed up?! :/
